app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -33,6 +34,7 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger("WaitActivity");
     private IServiceLocator serviceLocator;
+    private IDatabase database;
```

Comment:
To follow the law of Demeter, that is, you should only communicate with your friends, not with friends of friends. I think that the law of Demeter really makes code a lot less error-prone and reduces the change on null-pointers exponentially. A class is a friend if it's in a field of the class.